vulnerable_people_form/integrations/persistence.py

```diff
@@ -22,8 +23,16 @@ def get_client_kwargs(app=current_app):
     }
 
 
+def set_connection_header(request, operation_name, **kwargs):
+    request.headers['Connection'] = 'keep-alive'
+
+
 def get_rds_data_client(app=current_app):
-    return boto3.client("rds-data", **get_client_kwargs(app))
+    if 'boto_rds_client' not in g:
```

Comment:
this is really neat! I presume its more reliable than just instantiating a global client on import?

Reply:
Just going by documentation on how to maintain db connections. https://flask.palletsprojects.com/en/1.1.x/appcontext/

You could have some retry logic here if the client gets in a bad state.

Reply:
cool stuff!

Reply:
what happens if it does get a bad state and the app is still running ? Is that something we can pre-empt?